src/vm/core_test.zig

```diff
@@ -867,6 +870,200 @@ test "deduceOp1 when opcode == .AssertEq, res_logic == .Op1, no op0" {
+    // ************************************************************
+    var instr = deduceOpTestInstr;
+    instr.opcode = .Call;
+
+    const tuple = try vm.deduceOp0(&instr, null, null);
+    const op1 = tuple[0];
+    const res = tuple[1];
+
+    // ************************************************************
+    // *                      TEST CHECKS                         *
+    // ************************************************************
+    const expectedOp0: ?MaybeRelocatable = relocatable.newFromRelocatable(relocatable.Relocatable.new(0, 1)); // temp var needed for type inference
+    const expectedRes: ?MaybeRelocatable = null;
+    try expectEqual(expectedOp0, op1);
+    try expectEqual(expectedRes, res);
+}
+
+test "deduceOp0 when opcode == .AssertEq, res_logic == .Add, input is felt" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    var vm = try CairoVM.init(gpa_allocator, .{});
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    var instr = deduceOpTestInstr;
+    instr.opcode = .AssertEq;
+    instr.res_logic = .Add;
+
+    const dst = relocatable.fromU64(3);
+    const op0 = relocatable.fromU64(2);
+
+    const tuple = try vm.deduceOp0(&instr, &dst, &op0);
+    const op1 = tuple[0];
+    const res = tuple[1];
+
+    // ************************************************************
+    // *                      TEST CHECKS                         *
+    // ************************************************************
+    try expect(op1.?.eq(relocatable.fromU64(1)));
+    try expect(res.?.eq(relocatable.fromU64(3)));
+}
+
+test "deduceOp0 when opcode == .AssertEq, res_logic == .Add, with no input" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    var vm = try CairoVM.init(gpa_allocator, .{});
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    var instr = deduceOpTestInstr;
+    instr.opcode = .AssertEq;
+    instr.res_logic = .Add;
+
+    const tuple = try vm.deduceOp0(&instr, null, null);
+    const op1 = tuple[0];
+    const res = tuple[1];
+
+    // ************************************************************
+    // *                      TEST CHECKS                         *
+    // ************************************************************
+    const expectedOp0: ?MaybeRelocatable = null; // temp var needed for type inference
+    const expectedRes: ?MaybeRelocatable = null;
```

Comment:
```suggestion
    const expected_op_0: ?MaybeRelocatable = null; // temp var needed for type inference
    const expected_res: ?MaybeRelocatable = null;
```

camelCase only for functions, snake_case for others.